CanHazFunny/CanHazFunny/JokeService.cs

```diff
@@ -1,14 +1,26 @@
-﻿using System.Net.Http;
-
+using System.Net.Http;
 namespace CanHazFunny;
 
-public class JokeService
-{
-    private HttpClient HttpClient { get; } = new();
-
-    public string GetJoke()
+    public partial class JokeService : IJokeService
     {
-        string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
-        return joke;
-    }
+        private HttpClient HttpClient { get; } = new();
+        public string GetJoke()
+        {
+            HttpResponseMessage json = HttpClient.GetAsync("https://geek-jokes.sameerkumar.website/api?format=json").Result;
+
+            string joke = ParseJokeJSON(json.Content.ReadAsStringAsync().Result);
+
+            return joke;
```

Comment:
Not the worst way to parse json, but probably harder than it needs to be. I think something like this would be simpler
```suggestion
         string json = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api?format=json").Result;
        var joke = JsonSerializer.Deserialize<JokeInformation>(json);
        if (joke != null && joke.Joke != null) return joke.Joke;
        throw new System.Exception("No joke received from API or joke was null");
```